lib/errors.ts

```diff
@@ -412,6 +412,14 @@ export class BalenaRequestError extends BalenaError {
 		public body: string,
 		public statusCode: number,
 		public requestOptions: object,
+		public responseHeaders?: {
+			get(name: string): string | null;
+			has(name: string): boolean;
+			forEach(callbackfn: (value: string, key: string) => void): void;
+			entries(): IterableIterator<[string, string]>;
+			keys(): IterableIterator<string>;
+			values(): IterableIterator<string>;
+		},
```

Comment:
So that we are not bound to a specific Header type (node/browser) and the typing make it clear that you are only supposed to manipulate those headers.